Hyprlinkr/DefaultRouteDispatcher.cs

```diff
@@ -108,5 +95,73 @@ public string RouteName
         {
             get { return this.routeName; }
         }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase", Justification = "This method should produce URIs with lower-case letters, so ultimately, it would have to invoke some sort of ToLower method.")]
+        private Rouple ExtractRoupleFromController(
+            MethodCallExpression callExpression, 
+            IDictionary<string, object> routeValues)
+        {            
+            var controllerType = callExpression.Object.Type;
+                        
+            var controllerRouteAttribute = controllerType.GetCustomAttribute<RouteAttribute>(false);
+            if (controllerRouteAttribute != null)
+            {
+                return ExtractRoupleFromControllerRouteAttribute(callExpression, routeValues, controllerRouteAttribute);
+            }
+
+            var controllerName = controllerType.Name.ToLowerInvariant().Replace("controller", "");
+
+            var newRouteValues = new Dictionary<string, object>(routeValues);
+            newRouteValues["controller"] = controllerName;
+
+            return new Rouple(this.routeName, newRouteValues);
+        }
+
+        private Rouple ExtractRoupleFromControllerRouteAttribute(
+            MethodCallExpression callExpression, 
+            IDictionary<string, object> routeValues, 
+            RouteAttribute controllerRouteAttribute)
+        {
+            var controllerRouteName = this.routeName;
+            if (controllerRouteAttribute.Name != null)
+            {
+                controllerRouteName = controllerRouteAttribute.Name;
+            }
+
+            var newRouteValues = AddControllerNameAndMethodToRouteValues(callExpression, routeValues);
+            newRouteValues = RemoveControllerAndActionRouteValuesIfNeeded(controllerRouteAttribute, newRouteValues);
```

Comment:
It's not that I want to be a killjoy here, but I can remove this method call without any tests failing.

It's not that I predict that _a lot_ of work will be done on Hyprlinkr in the future, but I tend to rely heavily on unit tests when I refactor. My modus operandi is that if all tests are still green after a refactoring, then I broke nothing.

There's a risk, then, that I (or someone else) might break a feature for you in the future, because this isn't covered by tests. If this is important to you, would it be possible to add some tests? Or perhaps remove this for now, and then come back to it in a later pull request?